src/core/defs.jl

```diff
@@ -547,9 +547,82 @@ the list unless one of the keywords, `first`, `last`, `before`, `after`.
 """
 function replace_comp!(md::ModelDef, comp_id::ComponentId, comp_name::Symbol=comp_id.comp_name;
                            first::VoidInt=nothing, last::VoidInt=nothing,
-                           before::VoidSymbol=nothing, after::VoidSymbol=nothing)
-    delete!(md, comp_name)
+                           before::VoidSymbol=nothing, after::VoidSymbol=nothing,
+                           reconnect::Bool=true)
+
+    if ! haskey(md.comp_defs, comp_name)
+        error("Cannot replace '$comp_name'; component not found in model.")
+    end
+
+    # Get original position if new before or after not specified
```

Comment:
Is this commented out because it's WIP? Add "TODO:" to the comments to indicate this. In general, we try to avoid checking in commented out code since it's easy to lose track of when the author intends to use it vs was just nervous about deleting it. (We have git for that!)

Reply:
yes this is still WIP. I put a todo list above in the comments, but I'll start adding it in the code as well!